doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+        Moniker = 3,
+        Command = 4,
+        Tag = 5,
+        /// DESIGN NOTE: The following PackageMatchField from AppInstallerRepositorySearch.h are not implemented in V1.
+        /// PackageFamilyName,
+        /// ProductCode,
+        /// NormalizedNameAndPublisher,
+    };
+    /// IMPLEMENTATION NOTE: PackageMatchFilter from AppInstallerRepositorySearch.h
+    [version(1)]
+    struct PackageMatchFilter
+    {
+        /// Whether the Package must match the filter in order to be included in the matches.
+        Boolean IsAdditive;
+        /// The type of string comparison for matching
+        MatchType Type;
+        /// The field to search
+        PackageMatchField Field;
+        /// The value to match
+        String Value;
+        /// DESIGN NOTE: "Additional" from RequestMatch AppInstallerRepositorySearch.h is not implememented here.
+    };
+
+    /// IMPLEMENTATION NOTE: ResultMatch from AppInstallerRepositorySearch.h
+    /// A single result from the search.
+    runtimeclass ResultMatch
+    {
+        ResultMatch();
+
+        /// The package found by the search request.
+        Package Package { get; };
+
+        /// The highest order field on which the package matched the search.
+        PackageMatchFilter MatchCriteria { get; };
+    }
+    /// IMPLEMENTATION NOTE: SearchResult from AppInstallerRepositorySearch.h
+    /// Search result data returned from FindPackages
+    runtimeclass FindPackagesResult
+    {
+        FindPackagesResult();
+
+        /// The full set of results from the search.
+        Windows.Foundation.Collections.IVectorView<ResultMatch> Matches { get; };
+
+        /// If true, the results were truncated by the given SearchRequest::MaximumResults.
+        /// USAGE NOTE: Windows Package Manager does not support result pagination, there is no way to continue getting more results.
+        Boolean IsTruncated{ get; };
+    }
+
+    /// Options for FindPackages
+    runtimeclass FindPackagesOptions
+    {
+        FindPackagesOptions();
+
+        /// DESIGN NOTE: 
+        /// This class maps to SearchRequest from  AppInstallerRepositorySearch.h 
+        /// That class is a container for data used to filter the available manifests in an app catalog.
+        /// Its properties can be thought of as:
+        /// (Query || Inclusions...) && Filters...
+        /// If Query and Inclusions are both empty, the starting data set will be the entire database.
+        /// Everything && Filters...
+        /// That has been translated in this api so that 
+        /// Query is PackageMatchField::AppCatalogDefined and PackageMatchFilter.IsAdditive = true
+        /// Inclusions are PackageMatchFilter.IsAdditive = true
+        /// Filters are PackageMatchFilter.IsAdditive = false
+
+        /// Filters to find packages
+        /// USAGE NOTE: Only one filter with PackageMatchField::AppCatalogDefined is allowed.
+        Windows.Foundation.Collections.IVector<PackageMatchFilter> Filters;

```

Comment:
Collection-typed properties should be get-only. The caller says `options.Filters.Add(...)` which leaves the collection implementation under your control.